### PR TITLE
transmission-edit: Allow changes to torrent source flag

### PR DESCRIPTION
**Pull request by orbital-mango**
This PR addresses #3724. I would probably have preferred to use the same argument for the source in both transmission-add and transmission-edit utils but as '-r' was already taken I just went with the obvious '-s' for setting the source flag/tag instead.

### CODE REVIEW

File: utils/edit.cc [hunk 1]
```diff
@@ -360,6 +388,11 @@ int tr_main(int argc, char* argv[])
             changed |= replaceURL(&top, options.replace[0], options.replace[1]);
         }
 
+        if (options.source)
```
Comment (ckerr) [inline utils/edit.cc:391]:
```suggestion
        if (options.source != nullptr)
```

File: utils/edit.cc [hunk 2]
```diff
@@ -68,6 +70,10 @@ static int parseCommandLine(app_options& opts, int argc, char const* const* argv
             opts.replace[1] = optarg;
             break;
 
+        case 's':
```
Comment (ckerr) [inline utils/edit.cc:73]: This should be listed in `utils/transmission-edit.1` as well

**Review by ckerr — COMMENTED** [review]
Thanks!

File: utils/edit.cc (hunk 1 above)
Comment (orbital-mango) [reply utils/edit.cc:391]: Oh yes, well spotted

File: utils/edit.cc (hunk 2 above)
Comment (orbital-mango) [reply utils/edit.cc:73]: Ah I missed that too. Done.

**Comment by orbital-mango** [comment]
That seems a rather oddly unrelated test to fail and only on the macos-11-from-tarball check... WatchDir/WatchDirTest.retry/1 is the culprit. I hate to be that guy but it, er, works on my machine :smile:

**Review by ckerr — APPROVED** [review]

**Comment by ilf** [comment]
Yay, thank you @orbital-mango! <3

**Pull request merged**